app/router/student.py

```diff
@@ -1,384 +1,309 @@
-    enrollment_data = {}
-    for key in data.keys():
-        if key in mapping.ENROLLMENT_RECORD_PARAMS and key != "student_id":
-            enrollment_data[key] = data[key]
-    return enrollment_data
+def build_student_and_user_data(student_data):
+    data = {}
+    for key in student_data.keys():
+        if key in STUDENT_QUERY_PARAMS + USER_QUERY_PARAMS:
+            data[key] = student_data[key]
+    return data
 
 
-def build_student_data(data):
-    student_data = {}
-    for key in data.keys():
-        if key in mapping.STUDENT_QUERY_PARAMS:
-            student_data[key] = data[key]
-    return student_data
+def generate_JNV_student_id(data):
+    counter = settings.JNV_COUNTER_FOR_ID_GENERATION
 
+    if counter > 0:
+        JNV_Id = JNVIDGeneration(
+            data["region"], data["school_name"], data["grade"]
+        ).get_id
+        counter -= 1
+        return JNV_Id
 
-def build_user_data(data):
-    user_data = {}
-    for key in data.keys():
-        if key in mapping.USER_QUERY_PARAMS:
-            user_data[key] = data[key]
-    return user_data
-
-
-def id_generation(data):
-    if data["group"] == "JNVStudents":
-        counter = settings.JNV_counter
-        if counter > 0:
-            JNV_Id = JNVIDGeneration(
-                data["region"], data["school_name"], data["grade"]
-            ).get_id
-            counter -= 1
-            return JNV_Id
-        raise HTTPException(
-            status_code=400, detail="Student ID could not be generated. Max loops hit!"
-        )
-
-
-@router.get("/")
-def get_students(request: Request):
-    """
-    This API returns a student or a list of students who match the criteria(s) given in the request.
+    raise HTTPException(
+        status_code=400, detail="JNV Student ID could not be generated. Max loops hit!"
+    )
 
-    Optional Parameters:
-    student_id (str), birth_date (str), category (str), stream (str).
-    For extensive list of optional parameters, refer to the DB schema note on Notion.
 
-    Returns:
-    list: student data if student(s) whose details match, otherwise 404
+async def create_school_user_record(data, school_name):
+    school_data = school.get_school(build_request(query_params={"name": school_name}))
+    group_data = group.get_group(
+        build_request(query_params={"child_id": school_data["id"], "type": "school"})
+    )
 
-    Example:
-    > $BASE_URL/student/
-    returns [data_of_all_students]
+    await group_user.create_group_user(
+        build_request(
+            method="POST",
+            body={"group_id": group_data[0]["id"], "user_id": data["user"]["id"]},
+        )
+    )
+    enrollment_record_data = {
+        "academic_year": str(datetime.now().year)
+        + "-"
+        + str((datetime.now() + relativedelta(years=1)).year),
+        "is_current": "true",
+        "start_date": datetime.now().strftime("%Y-%m-%d"),
+        "end_date": "",
+        "group_id": school_data["id"],
```

Comment:
shouldn't this be group_data[0]

Reply:
Same thing. School table can have only one unique school, so the school API already returns it.